### DISCUSSION TITLE
Return more user stats via get-single-profile API request

### DISCUSSION THREAD

**Discussion by brootle** [deso-protocol/dips#221] [Medium CIPs]
Return more user stats via `get-single-profile` API request

Must include "follows" and "followers". 🔥

Right now any app that runs via DeSo API has to make 3 separate API calls to retrieve simple user information, to get number of followers and to get number of follows are 2 separate API calls. By simply including that data into `get-single-profile` will reduce number of API requests 3 times! This will dramatically increase the speed of DeSo Apps, right now it takes crazy amount of time to load user profiles that have a lot of followers, this can be easily seen if you check @Krassenstein profile with 18K followers.

`get-single-profile` should also return number of posts, comments, quotes, diamonds, likes and so on. This data can be easily aggregated inside user profile by simple counters that would increment each number by event.

I consider this update to be critical for scaling and user experience that would influence all existing and future social apps built on DeSo Protocol.

It must be done!🚀

**Comment by znmead** [comment]
Great optimization idea, brootle. fafb.

**Comment by Lux1n** [comment]
I’m not well versed enough to be able to provide a counter argument, so on the surface this sounds great!

what would be some valid pushback to @brootle’s proposal, @lazynina @maebeam @diamondhands0 @jackson-dean?

**Comment by sungkhum** [comment]
Yes, please!

**Comment by brootle** [comment]
@diamondhands0 can you share your thoughts on this? thanks!👍

**Comment by oO** [comment]
I've already modified /v1/block in my dev node to include the number of transactions in the block. I agree that separating counts from actually pulling the details is important.

@brootle don't let the fact that it's written in Go stop you. Setup a Dev Node, start hacking prototypes/solutions and you could even be submitting pull requests.

**Reply by brootle** [reply]
I know, I am just being lazy, so I have to play mastermind and tell others what should be done
